src/Forms/ConfirmedPasswordField.php

```diff
@@ -43,12 +46,26 @@ class ConfirmedPasswordField extends FormField
     public $requireStrongPassword = false;
 
     /**
-     * Allow empty fields in serverside validation
+     * Allow empty fields in when entering the password for the first time
+     * If this is set to true then a random password may be generated if the field is empty
+     * depending on the value of $self::generateRandomPasswordOnEmtpy
      *
      * @var boolean
      */
     public $canBeEmpty = false;
 
+    private bool $generateRandomPasswordOnEmpty = false;
+
+    /**
+     * Callback to generate a random password
+     *
+     * Note: cannot strongly type a callable property so need to use docblock typing instead
+     * https://wiki.php.net/rfc/typed_properties_v2
+     *
+     * @var callable|null
+     */
+    private $randomPasswordCallback = null;
```

Comment:
The link you provided also says
> The recommended workaround is to instead use the `Closure` type, in conjunction with `Closure::fromCallable()`. This ensures that the callable will remain callable independent of scope.

Should we be using `Closure` here?

Reply:
Yeah make sense, have updated